src/Core/PluginApplicationInterface.php

```diff
@@ -38,9 +38,20 @@ interface PluginApplicationInterface extends EventDispatcherInterface
      *
      * @param string|\Cake\Core\PluginInterface $name The plugin name or plugin object.
      * @param array $config The configuration data for the plugin if using a string for $name
+     * @param bool $optional If an exception for missing plugin should not be thrown.
      * @return $this
      */
-    public function addPlugin($name, array $config = []);
+    public function addPlugin($name, array $config = [], bool $optional = false);
+
+    /**
+     * Adds a plugin, but only when in CLI mode.
+     *
+     * @param string|\Cake\Core\PluginInterface $name The plugin name or plugin object.
+     * @param array $config The configuration data for the plugin if using a string for $name
+     * @param bool $optional If an exception for missing plugin should not be thrown.
+     * @return $this
+     */
+    public function addCliPlugin($name, array $config = [], bool $optional = false);
```

Comment:
I still don't see the need to this method. We already provide a `Application::bootstrapCli()` method in app skeleton where user's can load their CLI only plugins.

Reply:
We can also add a `--cli` flag to `PluginLoadCommand` to add the `addPlugin()` statement in above method instead of `bootstrap()`.